composer/loggers/progress_bar_logger.py

```diff
@@ -153,21 +155,15 @@ def __init__(
         self.train_pbar: Optional[_ProgressBar] = None
         self.eval_pbar: Optional[_ProgressBar] = None
 
-        if isinstance(console_log_level, str):
-            console_log_level = LogLevel(console_log_level)
+        log_to_console = log_to_console if (log_to_console is not None) else (not progress_bar)
+        console_log_level = LogLevel(console_log_level)
 
-        if log_to_console is None:
-            log_to_console = not progress_bar
+        def _should_log(state: State, log_level: LogLevel):
+            return (log_to_console and ((log_level < console_log_level) or  #type: ignore
```

Comment:
why the type ignore?

Reply:
this is also a bit unreadable to do it all in one line.
could we also allow `log_to_console` to be a lambda like before?

Reply:
I was trying to get PyRight to type check, but then it couldn't infer the `console_log_level` type, and so it was kind of a wash. I can revert it to a lambda... but then it has to be one line right?